src/controllers/review.js

```diff
@@ -0,0 +1,81 @@
+const dishModel = require('../models/dish');
+
+const CustomError = require('../utils/customError');
+
+const addReviewToDish = async (req, res, next) => {
+  const { dishId } = req.params;
+  const { rating, comment } = req.body;
+  const reviewData = { rating, comment };
+  try {
+    if (!rating) {
+      return next(new CustomError('Dish not found', 404));
+    }
+    const dish = await dishModel.findOne({ id: dishId });
+    if (!dish) {
+      return next(new CustomError('Dish not found', 404));
+    }
+    dish.reviews.push(reviewData);
+    await dish.save();
+    return res.status(201).json({
+      success: true,
+      data: dish,
+    });
+  } catch (err) {
+    return next(new CustomError(err.message, 422));
+  }
+};
+const updateReviewToDish = async (req, res, next) => {
+  const { dishId } = req.params;
+  const { rating, comment } = req.body;
+  try {
+    const dish = await dishModel.findOne({ id: dishId });
+    if (!dish) {
+      return next(new CustomError('Dish not found', 404));
+    }
+    const reviewUpdate = dishModel.findOne({
+      reviews: { $elemMatch: { id: req.user.id } },
+    });
+    if (!reviewUpdate) {
+      return next(new CustomError('Review not found', 404));
+    }
+    reviewUpdate.rating = rating;
+    reviewUpdate.comment = comment;
+    await dish.save();
+    return res.status(201).json({
+      success: true,
+      data: dish,
+    });
+  } catch (err) {
+    return next(new CustomError(err.message, 422));
+  }
+};
+const deleteReviewFromDish = async (req, res, next) => {
+  const { dishId } = req.params;
+  try {
+    const dish = await dishModel.findOne({ id: dishId });
+
+    if (!dish) {
+      return next(new CustomError('Dish not found', 404));
+    }
+    const reviewDelete = dishModel.findOne({
+      reviews: { $elemMatch: { id: req.user.id } },
+    });
+    if (!reviewDelete) {
```

Comment:
same thing here change the error message to: You don't have a review

Reply:
ok